src/array/cuda/fp16.cuh

```diff
@@ -5,61 +5,98 @@
  * \note this file is modified from TVM project:
  *       https://github.com/apache/tvm/blob/e561007f0c330e3d14c2bc8a3ef40fb741db9004/src/target/source/literal/cuda_half_t.h.
  */
-#ifndef DGL_ARRAY_FP16_CUH_
-#define DGL_ARRAY_FP16_CUH_
+#ifndef DGL_ARRAY_CUDA_FP16_CUH_
+#define DGL_ARRAY_CUDA_FP16_CUH_
 
 
 #ifdef USE_FP16
 #include <cuda_fp16.h>
+#include <algorithm>
 
-static __device__ __forceinline__ half max(half a, half b)
-{
+static __device__ __forceinline__ half max(half a, half b) {
 #if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 530
   return __hgt(__half(a), __half(b)) ? a : b;
 #else
-  return __half(max(float(a), float(b)));
+  return __half(max(float(a), float(b)));  // NOLINT
```

Comment:
Thanks for polishing the code quality.
Not a blocking comment: can you help me understand why there are so many NOLINT exception here?

Reply:
cpplint will complain you should use `static_cast<float>()` instead of C-style `float()` for casting. I think it's OK for intrinsics in `src/array/cuda/atomic.cuh` and `src/array/cuda/fp16.cuh` as it follows the style in CUDA programming guide.